contracts/pcv/morpho/MorphoCompoundPCVDeposit.sol

```diff
@@ -0,0 +1,119 @@
+/// @dev PCV Guardian functions withdrawERC20ToSafeAddress and withdrawAllERC20ToSafeAddress
+/// will not work with removing Morpho Tokens on the Morpho PCV Deposit because Morpho
+/// has no concept of mTokens. This means if the contract is paused, or an issue is
+/// surfaced in Morpho and liquidity is locked, Volt will need to rely on social
+/// coordination with the Morpho team to recover funds.
+contract MorphoCompoundPCVDeposit is PCVDeposit {
+    using SafeERC20 for IERC20;
+
+    /// @notice reference to the lens contract for morpho-compound v2
+    address public constant LENS = 0x930f1b46e1D081Ec1524efD95752bE3eCe51EF67;
+
+    /// @notice reference to the morpho-compound v2 market
+    IMorpho public constant MORPHO =
+        IMorpho(0x8888882f8f843896699869179fB6E4f7e3B58888);
+
+    /// @notice reference to underlying token
+    address public immutable token;
+
+    /// @notice cToken in compound this deposit tracks
+    /// used to inform morpho about the desired market to supply liquidity
+    address public immutable cToken;
+
+    constructor(
+        address _core,
+        address _cToken,
+        address _underlying
+    ) CoreRef(_core) {
+        cToken = _cToken;
+        token = _underlying;
+    }
+
+    /// @notice Returns the distribution of assets supplied by this contract through Morpho-Compound.
+    /// @return sum of suppliedP2P and suppliedOnPool for the given CToken
+    function balance() public view override returns (uint256) {
+        (, , uint256 totalSupplied) = ILens(LENS).getCurrentSupplyBalanceInOf(
+            cToken,
+            address(this)
+        );
+
+        return totalSupplied;
+    }
+
+    /// @notice returns the underlying token of this deposit
+    function balanceReportedIn() external view returns (address) {
+        return token;
+    }
+
+    /// @notice deposit ERC-20 tokens to Morpho-Compound
+    function deposit() public whenNotPaused {
+        uint256 amount = IERC20(token).balanceOf(address(this));
+        if (amount == 0) {
+            /// no op to prevent revert on empty deposit
+            return;
+        }
+
+        IERC20(token).approve(address(MORPHO), amount);
+        MORPHO.supply(
+            cToken, /// cToken to supply liquidity to
+            address(this), /// the address of the user you want to supply on behalf of
+            amount
+        );
+
+        emit Deposit(msg.sender, amount);
+    }
+
+    /// @notice withdraw tokens from the PCV allocation
+    /// @param to the address PCV will be sent to
+    /// @param amount of tokens withdrawn
+    function withdraw(address to, uint256 amount) external onlyPCVController {
+        _withdraw(to, amount);
+    }
+
+    /// @notice withdraw all tokens from Morpho
+    /// @param to the address PCV will be sent to
+    function withdrawAll(address to) external onlyPCVController {
+        uint256 amount = balance();
+        _withdraw(to, amount);
+    }
```

Comment:
The `amount` here is computed through the Lens and it might be off a bit because of rounding errors. Indeed, the Lens does not assume that the correponding indexes are up to date (even if they are, because of the `withdraw` interaction with Morpho), and has to recompute them according to old indexes.

If the `amount` is slightly higher, `MORPHO.withdraw` will go through because it does a `min` with the maximum withdrawable amount, but the `safeTransfer` might revert.

A solution might be to compute the difference of ERC20 balance before and after a call to `MORPHO.withdraw(cToken, type(uint256).max)`, and use that as the amount to transfer to `to`. The parameter `type(uint256).max` is useful here to ensure that there are no dust left on Morpho. If the ERC20 balance of this contract is expected to always be 0, then it might be ok to only use the balance after the call to `MORPHO.withdraw` as the amount to transfer

Reply:
We discussed this internally, we could also call updateP2PIndexes to accrue interest so the balance function returns the proper balance with all interest accrued. It would only add an additional call from a gas cost perspective and would explicitly state all the assumptions.